src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs

```diff
@@ -318,16 +316,16 @@ private async Task<string> GetImdsUriApiVersionAsync(ILoggerAdapter logger, Dict
             Uri imdsErrorUri = new(ImdsEndpoint);
 
             HttpResponse response = await _httpManager.SendRequestAsync(
-                  imdsErrorUri,
-                  headers,
-                  body: null,
-                  HttpMethod.Get,
-                  logger: logger,
-                  doNotThrow: false,
-                  mtlsCertificate: null,
-                  customHttpClient: null,
-                  GetCancellationToken(userCancellationToken))
-                     .ConfigureAwait(false);
+                    imdsErrorUri,
+                    headers,
+                    body: null,
+                    method: HttpMethod.Get,
+                    logger: logger,
+                    doNotThrow: false,
+                    mtlsCertificate: null,
+                    validateServerCertificate: null,
```

Comment:
The original name `customHttpClient` meant that the app developer was customizing the httpClient. The new name can feel narrower, like it’s only about SSL certificate validation.

Reply:
No the original name meant that we are using a custom client with validation. It was not passed by app developer. The new name is not for the http client but the function to validate the server cert. I am not sure what you are suggesting here.

Reply:
you can see [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/d88df713764f0c1aca48f4e74d878993051c5843/src/client/Microsoft.Identity.Client/Http/HttpManager.cs#L155) for the current implementation. 

Basically, 

Case A: Both a certificate and a custom client? → Throw an exception

Case B: A custom client only? → Use the caller’s provided HttpClient as-is.

Case C: No custom client, but the factory supports mutual TLS? → Build an HttpClient preconfigured with the certificate.

Case D: No custom client, no specialized MTLS factory? → Fall back to the factory’s default HttpClient.

Now, with the new naming, seems like we lost the essence of that case B.

Reply:
I don't think we lost the essence. The callback needs a new handler and so does the mtls cert. As long as I know we do not have this support on SF so I do not consider this as a blocking comment.